File: app/src/main/java/ru/strcss/projects/moneycalc/moneycalcandroid/componets/spendingsections/SpendingSectionsContract.java
package ru.strcss.projects.moneycalc.moneycalcandroid.componets.spendingsections;

import java.util.List;

import ru.strcss.projects.moneycalc.moneycalcandroid.BasePresenter;
import ru.strcss.projects.moneycalc.moneycalcandroid.BaseView;
import ru.strcss.projects.moneycalc.moneycalcdto.entities.SpendingSection;

/**
 * This specifies the contract between the view and the loginPresenter.
 */
public interface SpendingSectionsContract {
    interface View extends BaseView<Presenter> {

        void showSpendingSections(List<SpendingSection> spendingSections);

        void showSpinner();

        void hideSpinner();

        void showUpdateSuccess();

        void showDeleteSuccess();
    }

    interface Presenter extends BasePresenter<View> {
        void requestSpendingSections();

        void deleteSpendingSection(Integer id);
    }
}

File: app/src/main/java/ru/strcss/projects/moneycalc/moneycalcandroid/componets/spendingsections/SpendingSectionsContract.kt
package ru.strcss.projects.moneycalc.moneycalcandroid.componets.spendingsections

import ru.strcss.projects.moneycalc.moneycalcandroid.BasePresenter
import ru.strcss.projects.moneycalc.moneycalcandroid.BaseView
import ru.strcss.projects.moneycalc.moneycalcdto.entities.SpendingSection

/**
 * This specifies the contract between the view and the loginPresenter.
 */
interface SpendingSectionsContract {
    interface View : BaseView<Presenter> {

        fun showSpendingSections(spendingSections: List<SpendingSection>)

        fun showSpinner()

        fun hideSpinner()

        fun showUpdateSuccess()

        fun showDeleteSuccess()
    }

    interface Presenter : BasePresenter<View> {
        fun requestSpendingSections()

        fun deleteSpendingSection(id: Int?)
    }
}
